_posts/2017-10-02-learning-from-8thlight.md

```diff
@@ -0,0 +1,74 @@
+
+## Learn about Apprenticeship
+
+After the great lunch (I don’t think you can eat bad in London) I sat aside Mollie and Katerina. Before lunch Katerina didn’t want to hook up the database yet to her Ruby code, since she expected to refactor the code a bit first. That’s what Mollie now helped Katerina with. They opened the tests she had written and dived deep into mocks, spies, fakes and the alikes. It was amazing to see how easily Mollie explained what they are and how the RSpec implementation of them ticks. She didn’t talk all the time. Mollie used most of the time to allow Katerina to think. It was amazing to see and just underlines what Daniel describes in [his blog post][4-tips-mentoring]: “a mentor is like being a tour guide […] go too fast and they won't take everything in; go too slow and they'll be bored”. Mollie did a great job I think. It was fun to watch the two.
+
+[4-tips-mentoring]: https://dev.to/d_ir/4-tips-for-mentoring-developers-894
+
+Later [Rabea] took the time to answer all of our questions about apprenticeship and surrounding things. She had [graduated][rabea-graduates] in August 2016. From her we learned all the things about the schedule of an apprentice and that the apprenticeship is modelled after a real software project, which means there are weekly iterations, there is a planning and retrospective, the tasks have to be estimated and you are responsible for handling your “project”, which is your apprenticeship.
+Let’s wind time back a little bit. It was in June 2015 that [Rabea wrote][rabea-becomes-developer] on her blog: “If you had told me a year ago that I would be working as a developer now, I wouldn’t have believed you. A developer? Me?”. She is a developer and not just that, she is going beyond and teaching us how to setup an apprenticeship program. The apprenticeship program is what other companies would call onboarding, but in a way more dedicated and focused way. You get to learn the technologies, techniques and mechanisms that make up professional software development. This is not a course you go to, sit on a bench, fall asleep on your table and hopefuly pass an exam at the end of it. No, not at all. This apprenticeship is a hands-on learning experience where you have a mentor and co-mentor assigned to you. They help you structuring your apprenticeship. You have regular checkins with them, you exchange learnings with other apprentices and everyone around in the office. And you graduate to become a crafter, which means you have learned all the basics. By no means you know all the things. But you surely know how to help yourself in all kinds of situations. Oh, right … there are not only technology skills involved, absolutely correct. There is soft skills, agile basics and communication skills involved. The apprenticeship is what 8th Light sets as the initial bar you have to want to take. If you don’t, you probably never even make it all the way to graduation.
```

Comment:
"you have a mentor and co-mentor assigned to you" we try very hard to not actually assign mentors to apprentices. It's (almost always) an opt-in thing. We have a list of crafters currently available to be a mentor, and as new potential hires come through, we try to match them up during the interview process. It isn't perfect, but the intent is to avoid assigning.